### PR TITLE
test ApiGroup names order

### PR DESCRIPTION
**Pull request by myakove**
## Summary by CodeRabbit

- **New Features**
	- Reordered and reintroduced API group constants for improved organization.
	- Introduced a new test for verifying the alphabetical order of API group constants.

- **Tests**
	- Added a test function to ensure API group constants are sorted correctly.

- **Chores**
	- Updated the testing environment configuration to include a new environment for API group ordering tests.

### CODE REVIEW

**Comment by rh-bot-1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: tests/test_api_group_order.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    api_groups = api_group_class[0]
+    api_group_names = [node.target.id for node in api_groups.body]
+    sorted_api_group_names = sorted(api_group_names, key=lambda x: x.replace("_", " "))
+    errors: Set[str] = set()
+    for _diff in difflib.Differ().compare(api_group_names, sorted_api_group_names):
+        if _diff.startswith("-") or _diff.startswith("+") or _diff.startswith("?"):
+            _err_diff = _diff.split(" ")[-1]
+            errors.add(_err_diff)
+
```
[the remark this answers is not in the document]
Comment (myakove) [reply tests/test_api_group_order.py:26]: In the current implementation only the diff added to errors, this is what should be in errors

File: tests/test_api_group_order.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+def test_api_group_order():
+    with open("ocp_resources/resource.py") as fd:
+        content = fd.read()
+
+    tree = ast.parse(source=content)
+    api_group_class: List[ast.ClassDef] = [
+        node for node in ast.walk(tree) if isinstance(node, ast.ClassDef) and node.name == "ApiGroup"
+    ]
```
[the remark this answers is not in the document]
Comment (myakove) [reply tests/test_api_group_order.py:14]: Done

**Review by rnetser — APPROVED** [review]

**Pull request merged**